### PR TITLE
[REFACTOR] Remove isExplore + allow TimeSeriesQueryEditor to be used without defining explicitly default plugin kind (take the first)

### PR DESCRIPTION
**Pull request by celian-garcia**
# Description

First PR of a long list that would aim refactoring the explorer. 
Here I propose to remove ``isExplore`` toggle from the ``TimeSeriesQueryEditor`` plugin. 
It degrades slightly the design (see screenshot) but removes cumbersome code.
I think that we agree anyway on the fact that the explore should stop reuse it in favor of reuse some lower level components that needs to be created.

Waiting for me to redesign the explore, I can do this PR in a feature branch if we don´t want it to be in main like this. My opinion is it's not that uglier finally :sweat_smile: 

At the same time I realized that The ``TimeSeriesQueryEditor`` needed default plugin to be defined, where in my opinion, it could just take the first defined plugin as default if defaultPluginKinds is not set. @zhuje I can modify by the way the ``TraceQueryEditor`` if you want the same behavior. So you wouldn't have to define ``defaultPluginKinds`` theorically ^^'.

# Screenshots
image

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the
  following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `DOC`,`IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky.
  See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests)
  and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues
  include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

File: ui/explore/src/components/PanelEditor/PanelEditorForm.tsx [hunk 1]
```diff
@@ -12,28 +12,19 @@
...
   initialValues: PanelEditorValues;
   initialAction: Action;
-  exploreTitleComponent?: JSX.Element;
-}
-
-export interface PanelSpecEditorProps {
-  panelDefinition: PanelDefinition;
-  onQueriesChange: (queries: QueryDefinition[]) => void;
-  onPluginSpecChange: (spec: UnknownSpec) => void;
+  exploreTitleComponent?: React.JSX.Element;
```
Comment (jgbernalp) [inline ui/explore/src/components/PanelEditor/PanelEditorForm.tsx:27]:
```suggestion
  exploreTitleComponent?: React.ReactNode;
```

File: ui/prometheus-plugin/src/plugins/prometheus-time-series-query/PrometheusTimeSeriesQueryEditor.tsx [hunk 2]
```diff
@@ -59,41 +66,46 @@ export function PrometheusTimeSeriesQueryEditor(props: PrometheusTimeSeriesQuery
...
-      minStepPlaceholder={minStepPlaceholder}
-      minStep={minStep}
-      handleMinStepChange={handleMinStepChange}
-      handleMinStepBlur={handleMinStepBlur}
-    />
+    <Stack spacing={2}>
+      <FormControl margin="dense" fullWidth={false}>
+        {/* TODO: How do we ensure unique ID values if there are multiple of these? Can we use React 18 useId and
+                maintain 17 compatibility somehow with a polyfill/shim? */}
+        <InputLabel id="prom-datasource-label">Prometheus Datasource</InputLabel>
```
Comment (jgbernalp) [inline ui/prometheus-plugin/src/plugins/prometheus-time-series-query/PrometheusTimeSeriesQueryEditor.tsx:74]: Couldn't we use the datasource name?
```suggestion
        <InputLabel id={`prom-datasource-label-${datasource.name}`">Prometheus Datasource</InputLabel>
```

**Review by jgbernalp — CHANGES_REQUESTED** [review]

File: ui/prometheus-plugin/src/plugins/prometheus-time-series-query/PrometheusTimeSeriesQueryEditor.tsx (hunk 2 above)
Comment (celian-garcia) [reply ui/prometheus-plugin/src/plugins/prometheus-time-series-query/PrometheusTimeSeriesQueryEditor.tsx]: Updated with a variable computed this way 
```ts
const datasourceSelectLabelID = `prom-datasource-label-${selectedDatasource.name || 'default'}`;
```

File: ui/explore/src/components/PanelEditor/PanelEditorForm.tsx (hunk 1 above)
Comment (celian-garcia) [reply ui/explore/src/components/PanelEditor/PanelEditorForm.tsx]: updated

**Review by AntoineThebaud — APPROVED** [review]
LGTM! Nice cleanup 💯

**Pull request merged**